### PR TITLE
Disable to print color if `--no-color` is specified

### PR DESCRIPTION
**Pull request by idorax**
Pull Request Checklist

* [x] implement the feature
* [ ] write the documentation
* [x] extend the test coverage
* [ ] update the specification
* [ ] adjust plugin docstring
* [ ] modify the json schema
* [ ] mention the version
* [ ] include a release note

Fixes: #2759

### CODE REVIEW

File: tmt/utils.py [hunk 1]
```diff
@@ -2475,8 +2475,12 @@ def _indent(iterable: Iterable[str]) -> Iterator[str]:
             else:
                 for line in item.splitlines():
                     yield f'{INDENT * " "}{line}'
-
-    yield click.style(str(exception), fg='red')
+    # XXX: We should handle option `--no-color`
+    no_color = True # FIXME: should get the no_color from tmt/__main__.py
```
Comment (happz) [inline tmt/utils.py:2479]: Can you try the following? `apply_colors_logging` should tell you whether to apply colors or not. It uses params from the current click context, and uses the right function to evaluate all inputs.

```python
click_context = click.get_current_context(silent=True)
_, apply_colors_logging = tmt.log.decide_colorization(click_context.params['no_color'], click_context.params['force_color'])
```

Comment (idorax) [reply tmt/utils.py:2479]: Thanks very much for your quick feedback, @happz !

I had a try to get `click_context` in [render_exception()][01], but it is `None`. e.g.

```python
> /home/huanli/dev/2759/dev/tmt/utils.py(2480)render_exception()
-> click_context = click.get_current_context(silent=True)
(Pdb) l
2475 	            else:
2476 	                for line in item.splitlines():
2477 	                    yield f'{INDENT * " "}{line}'
2478 	    # XXX: We should handle option `--no-color`
2479 	    breakpoint()
2480 ->	    click_context = click.get_current_context(silent=True)
2481 	    _, apply_colors_logging = tmt.log.decide_colorization(
2482 	            click_context.params['no_color'],
2483 	            click_context.params['force_color'])
2484 	    #no_color = True  # FIXME: should get the no_color from tmt/__main__.py
2485 	    if not apply_colors_logging:
(Pdb) n
> /home/huanli/dev/2759/dev/tmt/utils.py(2481)render_exception()
-> _, apply_colors_logging = tmt.log.decide_colorization(
(Pdb) pp click_context
None
```

Looks we have to get `click_context` from [run_cli()][02]. What do you think?

[01]: https://github.com/teemtee/tmt/blob/main/tmt/utils.py#L2467
[02]: https://github.com/teemtee/tmt/blob/main/tmt/__main__.py#L4

Comment (happz) [reply tmt/utils.py:2479]: How did you run this code? Is it from a Python session, or from a genuine tmt command-line?

Comment (idorax) [reply tmt/utils.py:2479]: Hi @happz, I ran the code via a genuine tmt command-line and the related code having `breakpoint()`. e.g.

```python
(dev) huanli@KVM:dev$ git diff
diff --git a/tmt/utils.py b/tmt/utils.py
index 047713c5..0c3c114a 100644
--- a/tmt/utils.py
+++ b/tmt/utils.py
@@ -2476,8 +2476,13 @@ def render_exception(exception: BaseException) -> Iterator[str]:
                 for line in item.splitlines():
                     yield f'{INDENT * " "}{line}'
     # XXX: We should handle option `--no-color`
-    no_color = True # FIXME: should get the no_color from tmt/__main__.py
-    if no_color:
+    breakpoint()
+    click_context = click.get_current_context(silent=True)
+    _, apply_colors_logging = tmt.log.decide_colorization(
+            click_context.params['no_color'],
+            click_context.params['force_color'])
+    #no_color = True  # FIXME: should get the no_color from tmt/__main__.py
+    if not apply_colors_logging:
         yield str(exception)
     else:
         yield click.style(str(exception), fg='red')

(dev) huanli@KVM:dev$ tmt --no-color run discover -h fmf -u https://127.0.0.1/nonexistent
/var/tmp/tmt/run-001

/plans/features/advanced
    discover
        how: fmf
        url: https://127.0.0.1/nonexistent
        fail: Command 'git clone https://127.0.0.1/nonexistent /var/tmp/tmt/run-001/plans/features/advanced/discover/default-0/tests' returned 128.
        fail: Command 'git clone https://127.0.0.1/nonexistent /var/tmp/tmt/run-001/plans/features/advanced/discover/default-0/tests' returned 128.
        fail: Command 'git clone https://127.0.0.1/nonexistent /var/tmp/tmt/run-001/plans/features/advanced/discover/default-0/tests' returned 128.

> /home/huanli/dev/2759/dev/tmt/utils.py(2480)render_exception()
-> click_context = click.get_current_context(silent=True)
(Pdb) l
2475 	            else:
2476 	                for line in item.splitlines():
2477 	                    yield f'{INDENT * " "}{line}'
2478 	    # XXX: We should handle option `--no-color`
2479 	    breakpoint()
2480 ->	    click_context = click.get_current_context(silent=True)
2481 	    _, apply_colors_logging = tmt.log.decide_colorization(
2482 	            click_context.params['no_color'],
2483 	            click_context.params['force_color'])
2484 	    #no_color = True  # FIXME: should get the no_color from tmt/__main__.py
2485 	    if not apply_colors_logging:
(Pdb) n
> /home/huanli/dev/2759/dev/tmt/utils.py(2481)render_exception()
-> _, apply_colors_logging = tmt.log.decide_colorization(
(Pdb) p click_context
None
######## <<<<<<<<<<<<<<<<<<  !!! click_context is None !!!


(Pdb) n
> /home/huanli/dev/2759/dev/tmt/utils.py(2482)render_exception()
-> click_context.params['no_color'],
(Pdb)
AttributeError: 'NoneType' object has no attribute 'params'
> /home/huanli/dev/2759/dev/tmt/utils.py(2482)render_exception()
-> click_context.params['no_color'],
(Pdb)
```

Comment (happz) [reply tmt/utils.py]: Aha, got it. At this point, we left the Click code, which explains why Click cleaned up the context stack, and there is no active context :/ Now it makes sense.

Let me try a few experiments.

Comment (idorax) [reply tmt/utils.py]: Thanks very much for your solution, which is much better than mine. @happz

**Comment by idorax** [comment]
Hi @happz, please help complete the remaining work of this PR as I no longer have the permission to update it, thanks very much!

```bash
huanli@FOO:~/dev/2759/dev$ git remote -v
origin	git@github.com:teemtee/tmt.git (fetch)
origin	git@github.com:teemtee/tmt.git (push)

huanli@FOO:~/dev/2759/dev$ git branch -a | grep huanli
* dev-huanli-2759-20240326
  remotes/origin/dev-huanli-1062-20240320-fips
  remotes/origin/dev-huanli-1918-20230316-login-t
  remotes/origin/dev-huanli-2759-20240326

huanli@FOO:~/dev/2759/dev$ git push -f
ERROR: Permission to teemtee/tmt.git denied to idorax.
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.
```

**Comment by happz** [comment]
> Hi @happz, please help complete the remaining work of this PR as I no longer have the permission to update it, thanks very much!

That's weird, I don't recall any change that would suddenly prevent you from pushing to branches like you used to. Could it be just a temporary glitch? "fatal: Could not read from remote repository." suggests it might have been a network issue or Github outage or something like that. Have you tried pushing to `teemtee/tmt` since then?

I can definitely finish the PR if needed, but I'd prefer getting your workflow fixed so you could contribute even more :)

**Comment by idorax** [comment]
> hmm, I don't see @idorax in https://github.com/orgs/teemtee/people. Maybe it's best to just use a fork going forward.

Hi @martinhoyer, thank you very much for checking it. Yes, I'm not in the group. When I open link https://github.com/orgs/teemtee/people, such a message in the following is seen:

```
This organization has no public members. 
```

@happz, please help to  complete the remaining work of this PR, thank you very much!

**Comment by martinhoyer** [comment]
@idorax You can use fork branches :)

**Review by KwisatzHaderach — APPROVED** [review]

**Review by therazix — APPROVED** [review]

**Comment by happz** [comment]
Pushing out to 1.36, there seems to be failing tests as a side-effect of my changes.

**Pull request merged**